backend/src/services/users/users.service.ts

```diff
@@ -1,8 +1,21 @@
 import { prisma } from '@data/prisma-client';
 import { User } from '@prisma/client';
 
-const getUsers = async (): Promise<User[]> => {
-  return await prisma.user.findMany();
+const getUsers = async (username: string): Promise<User[]> => {
+  let data: User[];
+  if (username) {
+    data = await prisma.user.findMany({
+      where: {
+        name: {
+          contains: username,
+        },
+      },
+    });
+  } else {
+    data = await prisma.user.findMany();
+  }
+
+  return data;
```

Comment:
You don't need to create a variable here. Especially, with such a generic name that doesn't convey any context.

You can just use return before `findMany`

Reply:
I've read before that this is bad practice to use `return await`. That's why I did this.
@oleksandr-danylchenko, what do you think about?
https://kanby.medium.com/%D0%BF%D0%BE%D1%87%D0%B5%D0%BC%D1%83-%D0%B8%D1%81%D0%BF%D0%BE%D0%BB%D1%8C%D0%B7%D0%BE%D0%B2%D0%B0%D1%82%D1%8C-return-await-%D0%BF%D0%BB%D0%BE%D1%85%D0%B0%D1%8F-%D0%B8%D0%B4%D0%B5%D1%8F-e87b70015f0c

Reply:
That's right. We mustn't use `return await`. But, you should just return a promise. 
`return prisma.user...` is the expected return format. Also, it'll allow avoiding redundant if-else nesting

Reply:
I got it. Fixed.